modules/tests/shared/src/test/scala/lucuma/core/math/SignalToNoiseSuite.scala

```diff
@@ -18,12 +22,18 @@ final class SignalToNoiseSuite extends munit.DisciplineSuite {
 
   import ArbSignalToNoise.given
   import ArbSignalToNoise.bigDecimalSignalToNoise
+  import ArbSignalToNoise.posBigDecimalSignalToNoise
   import ArbSignalToNoise.stringSignalToNoise
 
-  checkAll("Order",                  OrderTests[SignalToNoise].order)
-  checkAll("JSON Codec",             CodecTests[SignalToNoise].codec)
-  checkAll("FromBigDecimalExact",    PrismTests(SignalToNoise.FromBigDecimalExact))
-  checkAll("FromBigDecimalRounding", FormatTests(SignalToNoise.FromBigDecimalRounding).formatWith(bigDecimalSignalToNoise))
-  checkAll("FromString",             FormatTests(SignalToNoise.FromString).formatWith(stringSignalToNoise))
+  checkAll("Order",                     OrderTests[SignalToNoise].order)
+  checkAll("JSON Codec",                CodecTests[SignalToNoise].codec)
+  checkAll("FromBigDecimalExact",       PrismTests(SignalToNoise.FromBigDecimalExact))
+  checkAll("FromBigDecimalRounding",
+    ValidSplitEpiTests(SignalToNoise.FromBigDecimalRounding).validSplitEpiWith(bigDecimalSignalToNoise))
+  checkAll("FromPosBigDecimalExact",    PrismTests(SignalToNoise.FromPosBigDecimalExact))
+  checkAll("FromPosBigDecimalRounding",
+    ValidSplitEpiTests(SignalToNoise.FromPosBigDecimalRounding).validSplitEpiWith(posBigDecimalSignalToNoise)
+  )
+  checkAll("FromString",                FormatTests(SignalToNoise.FromString).formatWith(stringSignalToNoise))
```

Comment:
Do the tests fail with regular `BigDecimal` and `PosBigDecimal` arbitraries?

Reply:
I've never used `ValidSplitEpi` and am not sure what it is, but if it is like `Format` I think we're good?  The tests each include:

* valid big decimals: in range and not overly precise
* "coverage" examples: big decimals that are in range but way too precise -- these can be rounded to be valid.  After the initial round trip the canonical format emerges.
* arbitrary big decimals which may or may not be in range

```scala
  private val validBigDecimalSignalToNoise: Gen[BigDecimal] =
    Gen
       .choose(1L, 9_999_999_999L)
       .map(BigDecimal(_, 3))

  private val coverageExamples: Gen[BigDecimal] =
     Gen
       .choose(1L, 9_999_999_999_999L)
       .map(BigDecimal(_, 6))

  val bigDecimalSignalToNoise: Gen[BigDecimal] =
     Gen.oneOf(validBigDecimalSignalToNoise, coverageExamples, arbitrary[BigDecimal])
```

Is that what you meant?

Reply:
My bad, I missed the fact that `bigDecimalSignalToNoise` includes `arbitrary[BigDecimal]` (and similarly `posBigDecimalSignalToNoise`), and so I thought we were leaving cases out. Sorry for that!